### PR TITLE
[Feature] Deferred callbacks

### PR DESCRIPTION
**Pull request by mikkoh**
When polling for `checkout.ready` we'll want to poll on a specific interval. This PR adds the ability to do deferred callbacks. This PR should be merged in after #75

### CODE REVIEW

File: scripts/generator/graphql_generator/csharp/SDK/UnityTimeout.cs.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public class UnityTimeout {
+        private static TimeoutComponent component;
+
+        public static void Start(float duration, Action callback) {
+            if (component == null) {
+                component = GlobalGameObject.AddComponent<TimeoutComponent>();    
+            }
+
+            component
+            .StartTimeout(duration, callback);
```
Comment (dengjeffrey) [inline scripts/generator/graphql_generator/csharp/SDK/UnityTimeout.cs.erb]: Nit: I think this line break is not necessary?

Comment (mikkoh) [inline scripts/generator/graphql_generator/csharp/SDK/UnityTimeout.cs.erb]: Ok I do not feel strongly. Will fix.

**Comment by mikkoh** [comment]
@dengjeffrey I've fixed the white space issue.

**Comment by mikkoh** [comment]
👏👏👏👏

**Pull request merged**